### PR TITLE
[GOMPS-451] Bugfix/cleanup unneeded coroutines

### PR DESCRIPTION
**Pull request by dwoodruffsf**
MLAPI has gotten better! Removed unneeded coroutines in ServerPostGameState and ServerCharSelectState. There is still an issue, GOMPS-444, somewhat related to this, but it happens regardless of whether the coroutine is present or not. That issue is separately documented as an [MLAPI bug](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/745). 

I tested a few cycles of dying and replaying with the host and one joiner, with 300ms RTT, and didn't have any problems.

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=287) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=287) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=287) it.</sub>

File: Assets/BossRoom/Prefabs/UI/PartyHUD.prefab [hunk 1]
```diff
@@ -515,12 +515,12 @@ MonoBehaviour:
   m_HeroPortrait: {fileID: 5210474332450112926}
```
Comment (fernando-cortez) [inline Assets/BossRoom/Prefabs/UI/PartyHUD.prefab:515]: At a glance, this seems like same change as the one done in PR 281. Does this change need to come through this PR?

File: Assets/BossRoom/Scripts/Server/Game/State/ServerPostGameState.cs [hunk 2]
```diff
@@ -23,15 +23,10 @@ public override void NetworkStart()
...
-                StartCoroutine(CoroSetWinLoss(won));
-            }
-        }
 
-        private IEnumerator CoroSetWinLoss(bool won)
-        {
-            yield return new WaitForSeconds(1);
-            GetComponent<PostGameData>().GameBannerState.Value =
-                (byte)(won ? PostGameData.BannerState.Won : PostGameData.BannerState.Lost);
+                GetComponent<PostGameData>().GameBannerState.Value =
```
Comment (fernando-cortez) [inline Assets/BossRoom/Scripts/Server/Game/State/ServerPostGameState.cs:27]: Can we serialize the `PostGameData` component and cache it?

**Review by SamuelBellomo — APPROVED** [review]

**Comment by Cosmin-B** [comment]
Other than what has been mentioned above, looks good to me

**Review by Cosmin-B — DISMISSED** [review]

File: Assets/BossRoom/Prefabs/UI/PartyHUD.prefab (hunk 1 above)
Comment (dwoodruffsf) [reply Assets/BossRoom/Prefabs/UI/PartyHUD.prefab:515]: yep, this diff has gone away, as you can see.

Comment (dwoodruffsf) [reply Assets/BossRoom/Prefabs/UI/PartyHUD.prefab:515]: it was needed for working on the PR, but the commit has now been taken into develop, and doesn't show up as a diff anymore.

File: Assets/BossRoom/Scripts/Server/Game/State/ServerPostGameState.cs (hunk 2 above)
Comment (dwoodruffsf) [reply Assets/BossRoom/Scripts/Server/Game/State/ServerPostGameState.cs]: you got it.

**Review by fernando-cortez — APPROVED** [review]

**Review by Cosmin-B — APPROVED** [review]

**Pull request merged**